src/server.py

```diff
@@ -78,6 +81,37 @@ def get_date_range(dates):
                 'endDate': max(dates).strftime('%Y-%m-%d %H:%M')}
 
 
+def valve(head_rev, pushlog_id):
+    """Determine which kind of job should been returned"""
+    low_job_return = True
+    COUNTER.increase_the_counter()
+    request_list = []
+    try:
+        request_list = session.query(TaskRequests.head_rev, TaskRequests.pushlog_id,
+                                     TaskRequests.low_job_return).all()
+    except Exception:
+        session.rollback()
+
+    requests = {}
+    for head_rev, pushlog_id, low_job_return in request_list:
+        requests[pushlog_id] = {'head_rev': head_rev,
+                                'low_job_return': low_job_return}
+    # If this pushlog_id has been schduled, we just return
+    # the priority returned before.
+    if pushlog_id in requests.keys():
+        low_job_return = requests.get(pushlog_id)['low_job_return']
+    else:
+        # we return all jobs for every 5 pushes.
+        if COUNTER.counter >= 5:
+            COUNTER.reset()
+            low_job_return = False
+        task_request = TaskRequests(str(head_rev), str(pushlog_id), low_job_return)
+        session.add(task_request)
+        session.commit()
+
+    return low_job_return
```

Comment:
'low_job_return' means 'we return the low job list for that push(True) or not(return all)'. I know it's a little weird in here. Could you give me some advices about it? Maybe rename it as priorities will better?